Bio/Phenotype/PhenMicro.py

```diff
@@ -0,0 +1,1101 @@
+    otherwise a nan value:
+    
+    >>> well[1.3]
+    23.0
+    >>> well[1250]
+    nan
+    
+    Two WellRecord objects can be compared: if their input time/signal pairs
+    are exactly the same, the two records are considered equal:
+    
+    >>> well2 = plate['H12']
+    >>> well == well2
+    False
+    
+    Two WellRecord objects can be summed up or subtracted from each other: a new
+    WellRecord object is returned, having the left operand id.
+    
+    >>> well2 = well + well1
+    >>> print(well2.id)
+    A05
+    
+    If SciPy is installed, a sigmoid function can be fitted to the PM curve,
+    in order to extract some parameters; three sigmoid functions are available:
+    * gompertz
+    * logistic
+    * richards
+    The functions are described in Zwietering et al., 1990 (PMID: 16348228)
+    
+    >>> well.fit()
+    >>> print(well.slope, well.model)
+    (61.853516785566917, 'logistic')
+    
+    If not sigmoid function is specified, the first one that is succesfully
+    fitted is used. The user can also specify a specific function.
+    
+    >>> well.fit('gompertz')
+    >>> print(well.slope, well.model)
+    (127.94630059171354, 'gompertz')
+    
+    If no function can be fitted, the parameters are left as None, except for
+    the max, min, average_height and area. 
+    """
+    
+    def __init__(self, wellid, plate=PlateRecord(None), signals={}):
+        self.plate = plate
+        
+        self.id = wellid
+    
+        # Curve parameters (to be calculated with the "fit" function)
+        # Parameters that don't need scipy
+        self.max = None
+        self.min = None
+        self.average_height = None
+        
+        # Parameters that need scipy
+        self.area = None
+        self.plateau = None
+        self.slope = None
+        self.lag = None
+        self.v = None
+        self.y0 = None
+        self.model = None
+    
+        # Original signals (private)
+        self._signals = signals
+    
+    def _interpolate(self, time):
+        """Private method to get a linear interpolation of the signals
+        at certain time points.
+        """
+        times = sorted(self._signals.keys())
+        
+        return np.interp(time,
+                        times,
+                        [self._signals[x] for x in times],
+                        left=np.nan, right=np.nan)
+    
+    def __setitem__(self, time, signal):
+        """Assign a signal at a certain time point.
+        """
+        try:
+            time = float(time)
+        except ValueError:
+            raise ValueError('Time point should be a number')
+        try:
+            signal = float(signal)
+        except ValueError:
+            raise ValueError('Signal should be a number')
+    
+        self._signals[time] = signal
+    
+    def __getitem__(self, time):
+        """Returns a subset of signals or a single signal.
+        """
+        if isinstance( time, slice ):
```

Comment:
PEP8 spacing would be ``if isinstance(time, slice):``

See also #493 for automating this checking.